frodo.js

```diff
@@ -1,131 +1,123 @@
 /**
```

Comment:
SCRIPT-Nice job on the comprehensive test coverage, it gives a lot of confidence in the stability of the new feature.